### ISSUE TITLE
v-for index as key on <transition-group> children with object properties

### ISSUE THREAD

**Issue by sylvainpolletvillard** [vuejs/eslint-plugin-vue#726]
Hey  guys, I recently updated vue-cli and got this new error: 
> Do not use v-for index as key on &lt;transition-group&gt; children, this is the same as not using keys.

But I don't really get it because my `v-for` is on an object, so indexes are property names. I can't think of a reason why object keys wouldn't be valid as a key, so I suppose it's a bug.

**Tell us about your environment**	
* **ESLint version:** 5.10.0	
* **eslint-plugin-vue version:** 5.0.0
* **Node version:**	10.11.0

**Please show your full configuration:**

```json
default config from @vue/cli
```

**What did you do?**	

```vue
 <ul>
        <transition-group name="list">
          <li
            class="category"
            v-for="(categoryCourses, category) in filteredCourses"
            :key="category"
          >
```

**What did you expect to happen?**
Here, `filteredCourses` is an object, not an array. So `category` is a property name and can be used as a key

**What actually happened?**

Error: Do not use v-for index as key on <transition-group> children, this is the same as not using keys.

**Comment by sylvainpolletvillard** [comment]
Just noticed that the warning comes from vue compiler, not eslint plugin: https://github.com/vuejs/vue/blob/d8285c57a613c42eddf2d4f2b75c1cea6aa4703a/src/compiler/parser/index.js#L343

Should I create the issue in `vue` repo instead ?

**Comment by armano2** [comment]
yes, but can you leave this issue open, we should add check for that

**Comment by edum18** [comment]
@sylvainpolletvillard I have the same issue. Did you create the issue already?

**Comment by sylvainpolletvillard** [comment]
@edum18 no, could you do it for me ? Thanks !

**Comment by edum18** [comment]
@sylvainpolletvillard sure mate

**Comment by edum18** [comment]
> @edum18 no, could you do it for me ? Thanks !

 [sure](https://github.com/vuejs/vue/issues/9215)

**Comment by sylvainpolletvillard** [comment]
My issue was specifically when using v-for on an object. The warning message makes sense when using array indexes.

**Comment by Holymaan** [comment]
<ul>
        <transition-group name="list" enter-active-class="animated bounceInUp" leave-active-class="animated bounceOutDown">
          <li v-for="(data, index) in skills" :key='index'>{{data.skill}}</li>
        </transition-group>
</ul>
I fixed this piece of code with this small adjustment
<li v-for="data in skills" :key='data'>{{data.skill}}</li>

**Comment by michalsnik** [comment]
Unfortunately we can't detect (and be 100% sure) whether list you're iterating over is an Array or Object.

Case with people wrongly using Array indexes as keys overweights inconvenience of turning off the rule per case where you iterate over Object entries, so we're going to leave it as is.

**Comment by jcrick69** [comment]
Folowing @sylvainpolletvillard 's idea I just did key='index + 0' and the error disappeared. Seems forcing a calculation gets over the problem...

**Comment by o8e** [comment]
A little more semantic...

`:key="`key-${index}`"`

**Comment by lk77** [comment]
Hello, 

same issue here, using index seems valid to me if the order of the elements in the array is never gonna change

**Comment by sudofox** [comment]
I don't have any kind of unique ID for the list items. Do I need to introduce randomized dummy data into each list item to overcome this deficiency?

**Comment by sergx** [comment]
I solved my problem with [object-hash](https://www.npmjs.com/package/object-hash). Made a mixin and now using:
```
<div v-for="line in grid_structure" :key="$hash(line)">
```
